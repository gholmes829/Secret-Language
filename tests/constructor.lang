# testing basic class constructors

class Cookie {
    void init(str topping1, str topping2) {
        this.topping1 = topping1
        this.topping2 = topping2
    }

    void eat() {
        print('Cookie with ' + this.topping1 + ' and ' + this.topping2)
    }

    void change_toppings(str topping1, str topping2) {
        this.topping1 = topping1
        this.topping2 = topping2
    }
}

num main() {
    let cookie = Cookie('chocolate', 'sprinkles')
    cookie.eat()

    cookie.change_toppings('pb', 'jelly')
    cookie.eat()

    return 0
}