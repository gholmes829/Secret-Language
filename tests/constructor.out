Cookie with chocolate and sprinkles
Cookie with pb and jelly
